weed/server/filer_server_handlers_read.go

```diff
@@ -105,7 +105,11 @@ func (fs *FilerServer) GetOrHeadHandler(w http.ResponseWriter, r *http.Request)
 		if err == filer_pb.ErrNotFound {
 			glog.V(2).Infof("Not found %s: %v", path, err)
 			stats.FilerHandlerCounter.WithLabelValues(stats.ErrorReadNotFound).Inc()
-			w.WriteHeader(http.StatusNotFound)
+			if r.Header.Get(s3_constants.XAmzBucketAccessDenied) == "true" {
```

Comment:
this s3 specific logic can be handled within s3api_object_handlers.go instead of leaking to filer.

Reply:
This change is aimed at reducing one file metadata lookup. If this logic were placed in `s3api_object_handlers.go`, the s3api would first need to fetch the metadata to perform the ACL check, and only after a successful check would the file download proceed. With this change, the file verification is passed down to the filer via request headers, so the s3api no longer needs to perform a separate metadata query for ACL checks during file download.